services/FacilityService.php

```diff
@@ -0,0 +1,283 @@
+        $sql .= "     country_code='" . $data["country_code"] . "',";
+        $sql .= "     federal_ein='" . $data["federal_ein"] . "',";
+        $sql .= "     website='" . $data["website"] . "',";
+        $sql .= "     email='" . $data["email"] . "',";
+        $sql .= "     color='" . $data["color"] . "',";
+        $sql .= "     service_location='" . $data["service_location"] . "',";
+        $sql .= "     billing_location='" . $data["billing_location"] . "',";
+        $sql .= "     accepts_assignment='" . $data["accepts_assignment"] . "',";
+        $sql .= "     pos_code='" . $data["pos_code"] . "',";
+        $sql .= "     domain_identifier='" . $data["domain_identifier"] . "',";
+        $sql .= "     attn='" . $data["attn"] . "',";
+        $sql .= "     tax_id_type='" . $data["tax_id_type"] . "',";
+        $sql .= "     primary_business_entity='" . $data["primary_business_entity"] . "',";
+        $sql .= "     facility_npi='" . $data["facility_npi"] . "',";
+        $sql .= "     facility_code='" . $data["facility_code"] . "'";
+        $sql .= " WHERE id='" . $data["fid"] . "'";
+
+        return sqlStatement($sql);
+    }
+
+    public function insert($data)
+    {
+        $sql  = " INSERT INTO facility SET";
+        $sql .= "     name='" . $data["name"] . "',";
+        $sql .= "     phone='" . $data["phone"] . "',";
+        $sql .= "     fax='" . $data["fax"] . "',";
+        $sql .= "     street='" . $data["street"] . "',";
+        $sql .= "     city='" . $data["city"] . "',";
+        $sql .= "     state='" . $data["state"] . "',";
+        $sql .= "     postal_code='" . $data["postal_code"] . "',";
+        $sql .= "     country_code='" . $data["country_code"] . "',";
+        $sql .= "     federal_ein='" . $data["federal_ein"] . "',";
+        $sql .= "     website='" . $data["website"] . "',";
+        $sql .= "     email='" . $data["email"] . "',";
+        $sql .= "     color='" . $data["color"] . "',";
+        $sql .= "     service_location='" . $data["service_location"] . "',";
+        $sql .= "     billing_location='" . $data["billing_location"] . "',";
+        $sql .= "     accepts_assignment='" . $data["accepts_assignment"] . "',";
+        $sql .= "     pos_code='" . $data["pos_code"] . "',";
+        $sql .= "     domain_identifier='" . $data["domain_identifier"] . "',";
+        $sql .= "     attn='" . $data["attn"] . "',";
+        $sql .= "     tax_id_type='" . $data["tax_id_type"] . "',";
+        $sql .= "     primary_business_entity='" . $data["primary_business_entity"] . "',";
+        $sql .= "     facility_npi='" . $data["facility_npi"] . "',";
+        $sql .= "     facility_code='" . $data["facility_code"] . "'";
+
+        return sqlInsert($sql);
+    }
+
+    /**
+     * Shared getter for the various specific facility getters.
+     *
+     * @param $map - Query information.
+     * @return array of associative arrays | one associative array.
+     */
+    private function get($map)
```

Comment:
Will it make sense to place this function(or part of it) in a separate class since I am guessing you will be using this same function in many other service classes in the future?